DEPENDENCIES.md

```diff
@@ -0,0 +1,58 @@
+# Dependency Management
+[`dep`](https://github.com/golang/dep) is a tool of choice when it comes to dependency management.
+
+## How we use `dep`.
+
+1. Transitive dependencies of `go-ethereum`. The most important thing for us is
+   to be in-sync there. We want to reduce the regression scope. 
+   Hence, we pin down all the dependencies of `go-ethereum` with SHAs in `Gopkg.toml` when
+   importing a new version of upstream. (This is considered a bad practice for
+   `dep` but we are willing
```

Comment:
It is based on an outdated version of #551. I think it is worth rebasing.

Reply:
I will cherry pick on develop when your change will be merged

Reply:
ah, ok. but maybe I should cherry-pick this one to my branch instead? so we don't add and then remove tons of files?

Reply:
good idea, i will close this PR